lua/obsidian/init.lua

```diff
@@ -91,6 +91,24 @@ obsidian.setup = function(opts)
       end
       cmp.setup.buffer { sources = sources }
     end
+
+    -- Keybindings...
+    -- 'gf' passthrough
+    if opts.keybindings.gf_passthrough then
+      local gf_set_by = vim.fn.mapcheck("gf", "n")
+      if gf_set_by == "" then
+        -- 'gf' has not been overridden by user or other plugin so we can safely set it.
+        vim.keymap.set("n", "gf", function()
+          if require("obsidian").util.cursor_on_markdown_link() then
```

Comment:
I think there should be an additional check here to ensure that the current buffer is located within the obsidian vault. Could cause problems if someone happens to be writing links formatted like Obsidian links elsewhere. 

Maybe it should be an optional parameter for `gf_passthrough`? `only_allow_in_vault` or something like that.

Reply:
Since we create this keymap within the lazy setup function and we have `buffer = true` in the keymap options, it's already only going to be set for buffers in the vault.